rotkehlchen/chain/ethereum/balancer/balancer.py

```diff
@@ -0,0 +1,321 @@
+            if asset_usd_price.value != Price(ZERO):
+                known_asset_prices[known_asset.address] = FVal(asset_usd_price.value)
+            else:
+                unknown_assets.add(known_asset.address)
+
+        return known_asset_prices
+
+    def _get_unknown_assets_prices(
+            self,
+            unknown_assets: Set[ChecksumEthAddress],
+    ) -> AssetsPrices:
+        """Get the tokens prices via the Balancer subgraph
+
+        Query
+        -----
+        Get list of tokenPrices filtering by token id
+
+        # ! TODO VN PR
+        # ! Add USD value per asset, either via PriceHistorian or The Graph
+        # ! Add Asset instance per asset, at least if exists
+        """
+        unknown_assets_prices: AssetsPrices = {}
+        # ! Format addresses, The Graph does not support checksum addresses
+        token_addresses = [address.lower() for address in unknown_assets]
+        param_types = {
+            '$token_ids': '[ID!]',
+        }
+        param_values = {
+            'token_ids': token_addresses,
+        }
+        result = self.graph.query(
+            querystr=TOKENPRICES_QUERY.format(),  # ! Remove returns in string
+            param_types=param_types,
+            param_values=param_values,
+        )
+        # ! Checksum address
+        unknown_assets_prices = {
+            to_checksum_address(token['id']): FVal(token['price'])
+            for token in result['tokenPrices']
+        }
+        return unknown_assets_prices
+
+    def _update_assets_prices_in_addresses_balancer_pools(
+            self,
+            addresses_balancer_pools: AddressesBalancerPools,
+            known_assets_prices: AssetsPrices,
+            unknown_assets_prices: AssetsPrices,
+    ) -> AddressesBalancerPools:
+        """Update the pools underlying assets prices in USD (prices obtained
+        via Inquirer and the Balancer subgraph)
+
+        Process highlights
+        ------------------
+        Per each <BalancerPoolAsset> in <BalancerPool>.assets:
+            * Replace <BalancerPoolAsset> including the asset USD price if it
+            exists either in `known_assets_prices` or `unknown_assets_prices`.
+            * Replace <BalancerPool> if any underlying asset has been replaced.
+        """
+        updated_addresses_balancer_pools: AddressesBalancerPools = {}
+        for address, balancer_pools in addresses_balancer_pools.items():
+            updated_balancer_pools: BalancerPools = []
+            for balancer_pool in balancer_pools:
+                updated_balancer_pool_assets: BalancerPoolAssets = []
+                is_updated = False
+                # Try to get price from either known or unknown assets prices.
+                # Otherwise keep existing price (zero)
+                # ! Assets prices values are <FVal> (truthy)
+                for asset in balancer_pool.assets:
+                    if asset.address in known_assets_prices:
```

Comment:
nitpick:
```python
asset_usd = known_assets_price.get(asset.address, unknown_assets_price.get(asset.address, ZERO))
```

is another way to write this. But perhaps your way is easier to understand.

Reply:
Promise it was my first approach, but then I thought it looked very weirdo split in multiple lines. But I will put it back!

Reply:
done